src/handlers/delegatedStakingHandler.ts

```diff
@@ -0,0 +1,192 @@
+import { PrismaClient } from "@prisma/client";
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
+import { extractHashOrdinal, extractPagination } from "../request-params";
+import {
+  paginatedQuery,
+  fromCreatedAtOrdinalCursor,
+  toCreatedAtOrdinalCursor,
+  toOrdinalCursor,
+  fromOrdinalCursor,
+} from "../pagination";
+import { handleError, respond } from "../response";
+import { toNumber } from "lodash";
+import { latestGlobalSnapshot } from "./dagHandler";
+
+const prisma = new PrismaClient();
+
+const delegateStakeCreateResponse = (event) => ({
+  hash: event.hash,
+  ordinal: event.ordinal,
+  source: event.source_addr,
+  nodeId: event.node_id,
+  amount: event.amount,
+  fee: event.fee,
+  tokenLockRef: event.token_lock_ref,
+  parentHash: event.parent_hash,
+  globalSnapshotHash: event.global_snapshot_hash,
+  timestamp: event.created_at,
+});
+
+const delegateStakeCreateResponses = (txs) =>
+  txs.map(delegateStakeCreateResponse);
+
+const delegateStakeWithdrawResponse = (event) => ({
+  hash: event.hash,
+  source: event.source_addr,
+  stakeRef: event.stake_ref,
+  globalSnapshotHash: event.global_snapshot_hash,
+  timestamp: event.created_at,
+});
+
+const delegateStakeWithdrawResponses = (txs) =>
+  txs.map(delegateStakeWithdrawResponse);
+
+const delegateStakeBalanceChangeResponse = (change) => ({
+  globalSnapshotOrdinal: change.global_snapshot_ordinal,
```

Comment:
I think we should hide `globalSnapshotOrdinal` for now. We don't plan to return anything except the latest balance at all times. We can add this later if we ever decide to allow historical queries.